exporter/elasticsearchexporter/README.md

```diff
@@ -177,7 +177,12 @@ See below for a description of each mapping mode.
 
 #### OTel mapping mode
 
-Requires Elasticsearch 8.12 or above.
+The default and recommended "OTel-native" mapping mode.
+
+Requires Elasticsearch 8.12 or above[^1], works best with Elasticsearch 8.16 or above[^2].
```

Comment:
For a user that uses the Elasticsearch exporter with Elasticsearch < 8.12 and the default mapping mode, they will stop seeing their data in Elasticsearch, and start seeing rejection logs in the collector. This is not ideal.

On the other hand, 8.12 is over a year old (January 2024) and the officially supported versions of Elasticsearch are currently 8.16 and 8.17.

I'm on board with this change as it makes new users' experience much better. Also the remediation is easy for affected users: add a configuration flag, as described in the changelog entry.